src/forces.f90

```diff
@@ -13,522 +13,832 @@
-         enddo
-       endif
-       icvlf_lx(iv) = icvlf(iv)
-       icvrt_lx(iv) = icvrt(iv)
-       jcvlw_lx(iv) = max(jcvlw(iv)+1-xstart(2),1)
-       jcvup_lx(iv) = min(jcvup(iv)+1-xstart(2),xsize(2))
-       icvlf_ly(iv) = max(icvlf(iv)+1-ystart(1),1)
-       icvrt_ly(iv) = min(icvrt(iv)+1-ystart(1),ysize(1))
-       jcvlw_ly(iv) = jcvlw(iv)
-       jcvup_ly(iv) = jcvup(iv)
-    enddo
-
-    if (nrank==0) then
-       write(*,*) '========================Forces============================='
-       write(*,*) '                       (icvlf)      (icvrt) '
-       write(*,*) '                (jcvup) B____________C '
-       write(*,*) '                        \            \ '
-       write(*,*) '                        \     __     \ '
-       write(*,*) '                        \    \__\    \ '
-       write(*,*) '                        \            \ '
-       write(*,*) '                        \       CV   \ '
-       write(*,*) '                (jcvlw) A____________D '
-       do iv=1,nvol
-          write(*,"(' Control Volume     : #',I1)") iv
-          write(*,"('     xld, icvlf     : (',F6.2,',',I6,')')") xld(iv), icvlf(iv)
-          write(*,"('     xrd, icvrt     : (',F6.2,',',I6,')')") xrd(iv), icvrt(iv)
-          write(*,"('     yld, jcvlw     : (',F6.2,',',I6,')')") yld(iv), jcvlw(iv)
-          write(*,"('     yud, jcvup     : (',F6.2,',',I6,')')") yud(iv), jcvup(iv)
-       enddo
-       write(*,*) '==========================================================='
-    endif
-
-    call decomp_2d_init_io(io_restart_forces)
-    call decomp_2d_register_variable(io_restart_forces, "ux01", 1, 0, 0, mytype)
-    call decomp_2d_register_variable(io_restart_forces, "uy01", 1, 0, 0, mytype)
-    call decomp_2d_register_variable(io_restart_forces, "ux11", 1, 0, 0, mytype)
-    call decomp_2d_register_variable(io_restart_forces, "uy11", 1, 0, 0, mytype)
-    
-  end subroutine init_forces
-
-  subroutine restart_forces(itest1)
-
-    USE decomp_2d_io
-    USE variables
-    USE param
-    USE MPI
-
-    implicit none
-
-    integer :: ierror,code,itest1
-    integer :: ierror_o=0 !error to open sauve file during restart
-    character(len=30) :: filename, filestart
-
-
+   use decomp_2d_constants
+   use decomp_2d_mpi
+   USE decomp_2d
+   implicit none
+ 
+   integer :: nvol,iforces,i2dsim
+   real(mytype),save,allocatable,dimension(:,:,:) :: ux01, uy01, ux11, uy11, ppi1
+   real(mytype),allocatable,dimension(:) :: xld,xrd,yld,yud,zfr,zbk
+   integer,allocatable,dimension(:) :: icvlf,icvrt,jcvlw,jcvup,kcvfr,kcvbk 
+   integer,allocatable,dimension(:) :: icvlf_lx,icvrt_lx,icvlf_ly,icvrt_ly
+   integer,allocatable,dimension(:) :: jcvlw_lx,jcvup_lx,jcvlw_ly,jcvup_ly
+   integer,allocatable,dimension(:) :: kcvfr_lx,kcvbk_lx,kcvfr_ly,kcvbk_ly
+ 
+   character(len=*), parameter :: io_restart_forces = "restart-forces-io", &
+        resfile = "restart-forces"
+   
+ contains
+ 
+   subroutine init_forces
+ 
+     USE decomp_2d
```

Comment:
This is not needed

Reply:
This is present in the latest version of the code itself.